hazelcast/src/main/java/com/hazelcast/internal/util/IntHolder.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright (c) 2008-2022, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.internal.util;
+
+import javax.annotation.concurrent.NotThreadSafe;
+
+/**
+ * A simple holder for a mutable int.
+ */
+@NotThreadSafe
+public class IntHolder {
```

Comment:
You can reuse `com.hazelcast.internal.util.MutableInteger`. But I also often use `int[1]` for this purpose.